filebeat/beater/channels.go

```diff
@@ -5,46 +5,43 @@ import (
 	"sync/atomic"
 
 	"github.com/elastic/beats/filebeat/input"
-	"github.com/elastic/beats/filebeat/registrar"
-	"github.com/elastic/beats/filebeat/spooler"
 )
 
-type spoolerOutlet struct {
-	wg      *sync.WaitGroup
-	done    <-chan struct{}
-	spooler *spooler.Spooler
-
+type prospectorOutput struct {
+	wg     *sync.WaitGroup
+	done   <-chan struct{}
+	input  chan *input.Event
 	isOpen int32 // atomic indicator
 }
 
-type publisherChannel struct {
+type spoolerOutput struct {
 	done chan struct{}
 	ch   chan []*input.Event
 }
 
-type registrarLogger struct {
+type publisherOutput struct {
 	done chan struct{}
 	ch   chan<- []*input.Event
 }
 
-type finishedLogger struct {
+type logger struct {
 	wg *sync.WaitGroup
 }
 
-func newSpoolerOutlet(
+func newProspectorOutput(
 	done <-chan struct{},
-	s *spooler.Spooler,
+	input chan *input.Event,
```

Comment:
while this one uses the spooler channel  only, idea of spoolerOutlet was to wrap/hide the spooler and the fact we're dealing with a channel. I'd even prefer in future calling a spooler.OnEvent directly, hiding the fact we're dealing with an asynchronous worker.

e.g. have a synchronous spooler worker with 'flush' upon full + potential async wrapper:

```
type spooler struct {}
type asyncSpooler struct {}

func (s *spooler) OnEvent()
func (s *spooler) Flush()

func newAsyncSpooler(s *spooler) *asyncSpooler
func (s *asyncSpooler) OnEvent()
func (s *asyncSpooler) Stop()

```

Both types support OnEvent, so no one needs to know about channels here.